### PR TITLE
fix(Scripts/BlackTemple) Gathios the Shatterer shouldnt use the same spell twice in a row

### PR DESCRIPTION
**Pull request by Carriola-wotlk**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/19667
- Closes https://github.com/chromiecraft/chromiecraft/issues/7280
- Closes https://github.com/chromiecraft/chromiecraft/issues/7261
- Closes https://github.com/chromiecraft/chromiecraft/issues/7250
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/19685

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
This behavior apparently happens for a few of Gathios's spells.
Auras: https://github.com/chromiecraft/chromiecraft/issues/7250
Seals: https://github.com/chromiecraft/chromiecraft/issues/7261
Blessings: https://github.com/chromiecraft/chromiecraft/issues/7280

I wonder though if the first of these spells he casts is random or always the same.

**Comment by amed80** [comment]
From what i have seen the first is always random

**Comment by Carriola-wotlk** [comment]
> This behavior apparently happens for a few of Gathios's spells. Auras: [chromiecraft/chromiecraft#7250](https://github.com/chromiecraft/chromiecraft/issues/7250) Seals: [chromiecraft/chromiecraft#7261](https://github.com/chromiecraft/chromiecraft/issues/7261) Blessings: [chromiecraft/chromiecraft#7280](https://github.com/chromiecraft/chromiecraft/issues/7280)
> 
you're right, now the question is: do you prefer that I fix all 3 bugs in this pr or open a pr for each one?


> From what i have seen the first is always random
ok, so I just need to ranodmize the initial _toggleIndex value

**Comment by heyitsbench** [comment]
> you're right, now the question is: do you prefer that I fix all 3 bugs in this pr or open a pr for each one?

Up to you, I'd do it all in one shot, but that's just me.

**Comment by sanctum32** [comment]
this could be handled with 1 variable

File: src/server/scripts/Outland/BlackTemple/illidari_council.cpp [hunk 1]
```diff
@@ -322,8 +322,10 @@ class boss_gathios_the_shatterer : public CreatureScript
             switch (events.ExecuteEvent())
             {
                 case EVENT_SPELL_BLESSING:
-                    if (Unit* member = SelectCouncilMember())
-                        me->CastSpell(member, RAND(SPELL_BLESSING_OF_SPELL_WARDING, SPELL_BLESSING_OF_PROTECTION), false);
+                    if (Unit* member = SelectCouncilMember()){
```
Comment (Nyeriah) [inline src/server/scripts/Outland/BlackTemple/illidari_council.cpp:325]: Curly to the new line

File: src/server/scripts/Outland/BlackTemple/illidari_council.cpp [hunk 2]
```diff
@@ -357,6 +359,12 @@ class boss_gathios_the_shatterer : public CreatureScript
 
             DoMeleeAttackIfReady();
         }
+        private:
+            uint32 _blessingSpells[2] = {
```
Comment (Nyeriah) [inline src/server/scripts/Outland/BlackTemple/illidari_council.cpp:363]: same here

**Comment by Carriola-wotlk** [comment]
I'll push the new changes soon that fix all three issues reported in these tickets:
Auras: https://github.com/chromiecraft/chromiecraft/issues/7250
Seals: https://github.com/chromiecraft/chromiecraft/issues/7261
Blessings: https://github.com/chromiecraft/chromiecraft/issues/7280

**Comment by Carriola-wotlk** [comment]
I pushed the new changes to fix these three issues:
Auras: https://github.com/chromiecraft/chromiecraft/issues/7250
Seals: https://github.com/chromiecraft/chromiecraft/issues/7261
Blessings: https://github.com/chromiecraft/chromiecraft/issues/7280

But I noticed that the boss Gathios can casts blessing on himself as well, but the strange thing is that I don't see the blessing aura applied to him and it's not logged in the combatLog.txt

Should I block this behavior?

File: src/server/scripts/Outland/BlackTemple/illidari_council.cpp [hunk 3]
```diff
@@ -284,7 +285,16 @@ class boss_gathios_the_shatterer : public CreatureScript
...
-        boss_gathios_the_shattererAI(Creature* creature) : boss_illidari_council_memberAI(creature) { }
+        boss_gathios_the_shattererAI(Creature* creature) : boss_illidari_council_memberAI(creature)
+        {
+            // Initialize toggle flags with random values to ensure the first toggled spell is cast randomly
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::bernoulli_distribution distrib(0.5);
+            _toggleBlessing = distrib(gen);
+            _toggleAura = distrib(gen);
+            _toggleSeal = distrib(gen);
```
Comment (Nyeriah) [inline src/server/scripts/Outland/BlackTemple/illidari_council.cpp:296]:
```suggestion
            _toggleBlessing = RAND(true, false);
            _toggleAura = RAND(true, false);
            _toggleSeal = RAND(true, false);
```

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**